ebos/eclgenericvanguard.cc

```diff
@@ -73,73 +70,56 @@ EclGenericVanguard::EclGenericVanguard()
+    EclGenericVanguard::setupTime_ = setupTime;
+    EclGenericVanguard::deck_ = std::move(deck);
+    EclGenericVanguard::eclState_ = std::move(eclState);
+    EclGenericVanguard::eclSchedule_ = std::move(schedule);
+    EclGenericVanguard::udqState_ = std::move(udqState);
+    EclGenericVanguard::actionState_ = std::move(actionState);
+    EclGenericVanguard::wtestState_ = std::move(wtestState);
+    EclGenericVanguard::eclSummaryConfig_ = std::move(summaryConfig);
 }
 
-void EclGenericVanguard::setExternalSchedule(std::unique_ptr<Schedule> schedule)
+void EclGenericVanguard::readDeck(const std::string& filename)
 {
-    externalEclSchedule_ = std::move(schedule);
-}
-
-void EclGenericVanguard::setExternalSummaryConfig(
-    std::shared_ptr<SummaryConfig> summaryConfig)
-{
-    externalEclSummaryConfig_ = std::move(summaryConfig);
-}
-
-void EclGenericVanguard::setExternalSummaryConfig(
-    std::unique_ptr<SummaryConfig> summaryConfig)
-{
-    externalEclSummaryConfig_ = std::move(summaryConfig);
-}
-
-void EclGenericVanguard::setExternalDeck(std::shared_ptr<Deck> deck)
-{
-    externalDeck_ = std::move(deck);
-    externalDeckSet_ = true;
-}
-
-void EclGenericVanguard::setExternalDeck(std::unique_ptr<Deck> deck)
-{
-    externalDeck_ = std::move(deck);
-    externalDeckSet_ = true;
-}
-
-void EclGenericVanguard::setExternalEclState(std::shared_ptr<EclipseState> eclState)
-{
-    externalEclState_ = std::move(eclState);
-}
-
-void EclGenericVanguard::setExternalEclState(std::unique_ptr<EclipseState> eclState)
-{
-    externalEclState_ = std::move(eclState);
-}
-
-void EclGenericVanguard::setExternalUDQState(std::unique_ptr<UDQState> udqState)
-{
-    externalUDQState_ = std::move(udqState);
-}
-
-void EclGenericVanguard::setExternalActionState(std::unique_ptr<Action::State> actionState)
-{
-    externalActionState_ = std::move(actionState);
-}
-
-void EclGenericVanguard::setExternalWTestState(std::unique_ptr<WellTestState> wtestState)
-{
-    externalWTestState_ = std::move(wtestState);
+    Dune::Timer setupTimer;
+    setupTimer.start();
+
+    std::shared_ptr<Opm::Deck> deck;
+    std::shared_ptr<Opm::EclipseState> eclipseState;
+    std::shared_ptr<Opm::Schedule> schedule;
+    std::unique_ptr<Opm::UDQState> udqState;
+    std::unique_ptr<Opm::Action::State> actionState;
+    std::unique_ptr<Opm::WellTestState> wtestState;
+    std::shared_ptr<Opm::SummaryConfig> summaryConfig;
+
+    auto parseContext =
+        std::make_unique<ParseContext>(std::vector<std::pair<std::string , InputError::Action>>
+                                            {{ParseContext::PARSE_RANDOM_SLASH, InputError::IGNORE},
+                                             {ParseContext::PARSE_MISSING_DIMS_KEYWORD, InputError::WARN},
+                                             {ParseContext::SUMMARY_UNKNOWN_WELL, InputError::WARN},
+                                             {ParseContext::SUMMARY_UNKNOWN_GROUP, InputError::WARN}});
```

Comment:
What about e.g. `ParseContext::PARSE_EXTRA_RECORDS` and strict parsing?

Reply:
This is only used in a couple of tests, and the code is just moved here. I guess it is not needed.